File: src/test/java/ROT13HiddenTest.java
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ROT13HiddenTest extends ROT13 {

    @Test
    public void EncryptTest() {
        assertEquals("NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm", encrypt("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"));
        assertEquals("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz", encrypt(encrypt("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")));
    }
}

File: src/test/scala/ROT13HiddenTest.scala
import org.scalatest._

class ROT13HiddenTest extends FlatSpec with Matchers {
  final val cipher = new ROT13

  "Encrypt" should "validate" in {
    cipher.encrypt("") shouldEqual ""
    cipher.encrypt(cipher.encrypt("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")) shouldEqual "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
    cipher.encrypt("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz") shouldEqual "NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm"
  }
}
